### PR TITLE
Improve LCM(-LoRA) Distillation Scripts

### PR DESCRIPTION
**Pull request by dg845**
# What does this PR do?

This PR makes further improvements to the LCM-(LoRA) distillation scripts.

Changelist:

1. For all scripts, make the interpolation function when rescaling configurable via the `--interpolation_type` argument.
2. For the LCM-LoRA scripts, make the `lora_alpha` (which controls LoRA scaling) and `lora_dropout` parameters configurable via the `--lora_alpha` and `--lora_dropout` arguments respectively.
3. For the LCM-LoRA scripts, the LoRA target modules can be made configurable via the `--lora_target_modules` argument.
4. For all scripts, make the VAE encoding batch size configurable via the `--vae_encode_batch_size` argument.
5. For all scripts, refactor the `scalings_for_boundary_conditions` function to use the `timestep_scaling` argument (instead of it being hardcoded to `10.0`) and make the scaling factor configurable via `--timestep_scaling_factor`.

Follow-up to #5778.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten
@sayakpaul
@patil-suraj

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
LGTM, thanks!

* Let's also support interpolation mode in the non-WDS script? 
* Might make sense to add `resolve_interpolation_mode()` in `training_utils.py`, as I imagine it being used in multiple scripts.

**Comment by patrickvonplaten** [comment]
@patil-suraj can you check here?

File: examples/consistency_distillation/train_lcm_distill_lora_sd_wds.py [hunk 1]
```diff
@@ -710,6 +724,50 @@ def parse_args():
...
+        default=0.0,
+        help="The dropout probability for the dropout layer added before applying the LoRA to each layer input.",
+    )
+    parser.add_argument(
+        "--lora_target_modules",
+        type=str,
+        default=None,
+        help=(
+            "A comma-separated string of target module keys to add LoRA to. If not set, a default list of modules will"
+            " be used."
```
Comment (patil-suraj) [inline examples/consistency_distillation/train_lcm_distill_lora_sd_wds.py:748]:
```suggestion
            " be used. By default, LoRA will be applied to all conv and linear layers."
```

File: examples/consistency_distillation/train_lcm_distill_lora_sdxl.py [hunk 2]
```diff
@@ -534,6 +545,50 @@ def parse_args():
...
+        default=0.0,
+        help="The dropout probability for the dropout layer added before applying the LoRA to each layer input.",
+    )
+    parser.add_argument(
+        "--lora_target_modules",
+        type=str,
+        default=None,
+        help=(
+            "A comma-separated string of target module keys to add LoRA to. If not set, a default list of modules will"
+            " be used."
```
Comment (patil-suraj) [inline examples/consistency_distillation/train_lcm_distill_lora_sdxl.py:569]: same comment as above

File: examples/consistency_distillation/train_lcm_distill_lora_sdxl_wds.py [hunk 3]
```diff
@@ -690,6 +704,50 @@ def parse_args():
...
+        default=0.0,
+        help="The dropout probability for the dropout layer added before applying the LoRA to each layer input.",
+    )
+    parser.add_argument(
+        "--lora_target_modules",
+        type=str,
+        default=None,
+        help=(
+            "A comma-separated string of target module keys to add LoRA to. If not set, a default list of modules will"
+            " be used."
```
Comment (patil-suraj) [inline examples/consistency_distillation/train_lcm_distill_lora_sdxl_wds.py:728]: same comment as above

**Review by patil-suraj — APPROVED** [review]
Very cool, lgtm!

**Comment by sayakpaul** [comment]
@dg845 it's not a blocker but do you want to carry over the changes to the non-WDS script as well?

**Comment by sayakpaul** [comment]
Thanks a lot, @dg845! The scripts are in a really spot now!

File: src/diffusers/training_utils.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 import numpy as np
 import torch
+from torchvision import transforms
```
Comment (yiyixuxu) [inline src/diffusers/training_utils.py:8]: we can't have torchvision as a required dependency

Comment (yiyixuxu) [reply src/diffusers/training_utils.py:8]: @DN6

**Comment by bigmover** [comment]
Wether TCD distilltion training is supported by Diffusers

**Pull request merged**